pkg/controller/controller_utils.go

```diff
@@ -549,6 +550,13 @@ func GetPodFromTemplate(template *v1.PodTemplateSpec, parentObject runtime.Objec
 	}
 	prefix := getPodsPrefix(accessor.GetName())
 
+	// if it is indexed job, add index information to its generateName prefix
+	if template.Annotations != nil {
+		if index, ok := template.Annotations[batch.JobCompletionIndexAnnotationAlpha]; ok {
```

Comment:
we need to change the annotation name, drop the Alpha suffix completely, better to do that in https://github.com/kubernetes/kubernetes/pull/101292

Reply:
then maybe hold on this PR until #101292 merged, so that the new annotation name can be used here

Reply:
changes of annotation name committed in #101292